app/assets/javascripts/hera/modules/combobox.js

```diff
@@ -0,0 +1,516 @@
+class ComboBox {
+  constructor($target) {
+    if (!$target.is('select')) {
+      console.error("Can't initialize a ComboBox on anything but a `select`");
+      return;
+    }
+
+    this.$target = $target;
+    this.config = (this.$target.data('combobox-config') || '')
+      .split(' ')
+      .filter(Boolean);
+
+    if (this.config.includes('no-combobox')) {
+      return;
+    }
+
+    this.debounceTimer = 250;
+    this.isMultiSelect = this.$target.attr('multiple');
+    this.idSuffix = Math.random().toString(36).substring(2);
+
+    this.init();
+  }
+
+  init() {
+    if (this.isAlreadyInitialized()) {
+      this.reinitialize();
+    }
+
+    this.buildDom();
+    this.setupFilter();
+    this.populateOptions();
+    this.setInitialSelection();
+    this.attachEventListeners();
+    this.setupMutationObserver();
+  }
+
+  appendCustomOption(value) {
+    const sanitizedValue = $('<div>').text(value).html();
+    this.$target.append(
+      `<option value="${sanitizedValue}">${sanitizedValue}</option>`
+    );
+
+    const $option = this.$target.children().last();
+
+    this.appendOption(this.$comboboxMenu, $option);
+
+    this.$comboboxOptions = this.$comboboxMenu.find(
+      '[data-behavior~=combobox-option]'
+    );
+
+    this.selectOptions(this.$comboboxOptions.last());
+  }
+
+  appendOption($parent, $option) {
+    const disabled = $option.attr('disabled') ? 'disabled' : '';
+    const escapedText = $('<div>').text($option.text()).html();
+    const value = $option.attr('value');
+
+    const {
+      comboboxOptionIcon: iconClass,
+      comboboxOptionIconColor: iconColor,
+      comboboxOptionColor: optionColor,
+    } = $option.data();
+
+    $parent.append(
+      `<span
+        aria-selected="false"
+        class="combobox-option ${disabled}"
+        data-behavior="combobox-option"
+        data-value="${value}"
+        id="combobox-option-${this.idSuffix}-${value}"
+        role="option"
+        tabindex="${disabled ? '-1' : '0'}"
+      >${escapedText}</span>`
+    );
+
+    const $appendedOption = $parent.children().last();
+
+    if (iconClass) {
+      $appendedOption.prepend(`<i class="${iconClass} fa-fw me-1"></i>`);
+
+      if (iconColor) {
+        $appendedOption.find('i').attr('style', `color: ${iconColor}`);
+      }
+    }
+
+    if (optionColor) {
+      $appendedOption.attr('style', `color: ${optionColor}`);
+    }
+
+    if (disabled) {
+      $appendedOption.attr('aria-disabled', 'true');
```

Comment:
why aria-disabled?

Reply:
This is to add support for accessibility when an option is disabled